allspice/utils/list_components.py

```diff
@@ -15,6 +15,7 @@
 PCB_FOOTPRINT_ATTR_NAME = "PCB Footprint"
 PART_REFERENCE_ATTR_NAME = "Part Reference"
 REPETITIONS_REGEX = re.compile(r"Repeat\(\w+,(\d+),(\d+)\)")
+MULTI_PART_NAME_REGEX = re.compile(r"(.)+\.([A-Z])")
```

Comment:
There's a couple issues with this regex. The first group should include the +. But it isn't used, as far as I can tell. So just remove it. Then it's followed by a dot. To prevent `.` matching literal `.`, which would lead to backtracking, one way is to use a lazy match.

```suggestion
MULTI_PART_NAME_REGEX = re.compile(r".+?\.([A-Z])")
```

If you know the first group cannot contain literal `.`, it might be best to match on that. Then you don't need lazy matching, which is simpler to reason about, in my opinion.

```suggestion
MULTI_PART_NAME_REGEX = re.compile(r"[^.]+\.([A-Z])")
```

Reply:
The first group can contain a `.`, but the lazy match is a great suggestion, thanks!